### PR TITLE
Wraeth/activity event separation [2/a]

### PR DESCRIPTION
**Pull request by wraeth-eth**
## What does this PR do and why?

_Provide a description of what this PR does. Link to any relevant GitHub issues, Notion tasks or Discord discussions._

## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the changes._

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

**Comment by wraeth-eth** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1772** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1772" target="_blank">Graphite</a> 
    * **PR #1777** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1777" target="_blank">Graphite</a> 
    * **PR #1895** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1895" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1895?utm_source=stack-comment).

File: src/components/activityEventElems/ActivityElement/ActivityElement.tsx [hunk 1]
```diff
@@ -34,43 +61,53 @@ const ExtraContainer: React.FC = ({ children }) => {
...
-  return (
-    <div style={smallHeaderStyle(colors)}>
-      <Trans>{header}</Trans>
-    </div>
-  )
+  return <div style={smallHeaderStyle(colors)}>{header}</div>
 }
 
-function SideDetails({ event }: { event: ActivityElementEvent }) {
+function SideDetails({ details }: { details: Partial<ActivityElementEvent> }) {
```
Comment (peripheralist) [inline src/components/activityEventElems/ActivityElement/ActivityElement.tsx:71]: little nit: can we use `Pick<>` to specify which props are needed, instead of the vague `Partial<>`

Comment (peripheralist) [inline src/components/activityEventElems/ActivityElement/ActivityElement.tsx:71]: little nit: can we use Pick<> to specify which props are needed, instead of the vague Partial<>

File: src/components/activityEventElems/ActivityElement/ActivityElement.tsx [hunk 2]
```diff
@@ -87,27 +124,177 @@ function Subject({ subject }: { subject: string | JSX.Element | null }) {
...
+
+export function ActivityEvent({
+  event,
+  style,
+}: {
+  event: ProjectEvent
+  style?: CSSProperties
+}) {
+  // TODO: we should use v2 where we can, or a better context
+  const { tokenSymbol } = useContext(V1ProjectContext)
```
Comment (peripheralist) [inline src/components/activityEventElems/ActivityElement/ActivityElement.tsx:162]: hmmm, we could actually store the symbol in the graph event

another context might also be a good solution, though i'd love to avoid scope creeping this PR any more lol

File: src/components/activityEventElems/ActivityElement/ActivityElement.tsx [hunk 3]
```diff
@@ -87,27 +124,177 @@ function Subject({ subject }: { subject: string | JSX.Element | null }) {
...
+        details: event.projectCreateEvent,
+      }
+    }
+    if (event.deployedERC20Event) {
+      return {
+        Header: DeployedERC20EventHeader,
+        Subject: (
+          <DeployedERC20EventSubject symbol={event.deployedERC20Event.symbol} />
+        ),
+        extra: null,
```
Comment (peripheralist) [inline src/components/activityEventElems/ActivityElement/ActivityElement.tsx:209]: should be `Extra` not `extra` eh? maybe we should type the return of this useMemo

File: src/components/v1/V1Project/ProjectActivity/index.tsx [hunk 4]
```diff
@@ -181,36 +173,25 @@ export default function ProjectActivity() {
...
 
-          if (e.payEvent) {
-            elem = <PayEventElem event={e.payEvent as PayEvent} />
+          if (
+            e.payEvent ||
+            e.redeemEvent ||
+            e.projectCreateEvent ||
+            e.deployedERC20Event
+          ) {
+            elem = <ActivityEvent event={e} />
```
Comment (peripheralist) [inline src/components/v1/V1Project/ProjectActivity/index.tsx:182]: LOOOOVE

wonder if there's a good way to move this `if (a || b || c)` logic to the ActivityEvent, so ProjectActivity doesn't have to care about which events have elements?

i guess if ActivityEvent renders `null`, then elem will == null?

File: src/components/v2/V2Project/ProjectActivity/index.tsx [hunk 5]
```diff
@@ -204,81 +191,20 @@ export default function ProjectActivity() {
...
-            >
-              {elem}
-            </div>
-          )
-        }),
-      ),
-    [colors, projectEvents],
+      projectEvents?.pages.flat().map(event => {
+        return (
+          <ActivityEvent
```
Comment (peripheralist) [inline src/components/v2/V2Project/ProjectActivity/index.tsx:196]: SO CLEAN

File: src/components/activityEventElems/ActivityElement/ActivityElement.tsx [hunk 6]
```diff
@@ -87,27 +124,177 @@ function Subject({ subject }: { subject: string | JSX.Element | null }) {
...
+  style,
+}: {
+  event: ProjectEvent
+  style?: CSSProperties
+}) {
+  // TODO: we should use v2 where we can, or a better context
+  const { tokenSymbol } = useContext(V1ProjectContext)
+
+  const { Header, Subject, Extra, details } = useMemo(() => {
+    if (event.payEvent) {
```
Comment (peripheralist) [inline src/components/activityEventElems/ActivityElement/ActivityElement.tsx:165]: wdyt about one more layer of organization... instead of importing a bunch of Header/Subject/Extra components here, we could replace this with some functions like `if (event.payEvent) return activityElemForPayEvent(event.payEvent)` where... `payEventElem()` returns a `{Header, Subject, Extra}` type

then the individual Header/Subject/Extra components could be contained in those files with only the main `activityElemFor...` function needing to be exported

**Comment by tomquirk** [comment]
@wraeth-eth should we close these 3 PRs?
